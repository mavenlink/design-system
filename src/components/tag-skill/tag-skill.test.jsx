import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import TagSkill from './tag-skill.jsx';

describe('TagSkill', () => {
  const requiredProps = {
    id: 'test-id',
    name: 'Egg',
  };

  it('has default', () => {
    expect(renderer.create(<TagSkill {...requiredProps} />).toJSON()).toMatchSnapshot();
  });

  describe('id API', () => {
    it('can be set', () => {
      render(<TagSkill {...requiredProps} id="unique-id" />);
      expect(screen.getByText('Egg').parentElement.parentElement).toHaveAttribute('id', 'unique-id');
    });
  });

  describe('name API', () => {
    it('can be set', () => {
      render(<TagSkill {...requiredProps} name="Bacon" />);
      expect(screen.getByText('Bacon')).toBeDefined();
    });
  });

  describe('level API', () => {
    it('can be set', () => {
      render(<TagSkill {...requiredProps} level={1} />);
      expect(screen.getByText('1')).toBeDefined();
    });
  });
});
