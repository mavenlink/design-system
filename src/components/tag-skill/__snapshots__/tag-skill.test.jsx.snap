// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagSkill has default 1`] = `
<div
  className="read-only-tag"
  id="test-id"
  role="row"
>
  <span
    className="content"
    id="test-id-content"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="gridcell"
    tabIndex="0"
  >
    <span
      className="row"
    >
      Egg
      <span
        className="level-tick"
      >
        <svg
          className="icon-base size-small"
          role="img"
        >
          <use
            xlinkHref="#icon-tick.svg"
          />
        </svg>
      </span>
    </span>
  </span>
</div>
`;
